### PR TITLE
fix service name for new registry

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: pkg/serviceaccounts/controllers/docker_registry_service.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ type serviceLocation struct {
 
 var serviceLocations = []serviceLocation{
 	{namespace: "default", name: "docker-registry"},
-	{namespace: "openshift-image-registry", name: "registry"},
+	{namespace: "openshift-image-registry", name: "image-registry"},
```
Comment (bparees) [inline pkg/serviceaccounts/controllers/docker_registry_service.go]: @deads2k why doesn't this controller/logic use the same registry hostname retriever logic as everything else?  (ie lookup the value from internalRegistryHostname, fall back to looking for existing services).

**Comment by bparees** [comment]
@adambkaplan fyi, one of many pieces that need fixing to get the image push/pulls/secrets aligned on "image-registry.openshift-image-registry.svc"

File: pkg/serviceaccounts/controllers/docker_registry_service.go (hunk 1 above)
Comment (deads2k) [inline pkg/serviceaccounts/controllers/docker_registry_service.go]: My pull adding this says "Makes the controller bilingual for both supported locations."

Are you sure you don't need to keep this?

**Comment by deads2k** [comment]
Distribution of trust so that the image registry can be used in the cluster seems like the domain of the image feature.  You want to add yourself to owners on this package at the same time?

File: pkg/serviceaccounts/controllers/docker_registry_service.go (hunk 1 above)
Comment (bparees) [reply pkg/serviceaccounts/controllers/docker_registry_service.go]: `docker-registry.default` is the "other supported location"

"registry.openshift-image-registry" is/was never a supported location

**Comment by bparees** [comment]
> You want to add yourself to owners on this package at the same time?

done

**Pull request merged**